.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main   
+  pull_request:  
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: clippy
+
+      - name: Install cargo-hack
+        run: cargo install cargo-hack
+
+      - name: Check compilation
+        run: cargo hack check --all-features --all-targets
+
+      - name: Check formatting
+        run: cargo +nightly fmt --all --check
```

Comment:
We'd have to keep formatting check in a separate job, since we didn't set up the nightly toolchain in this job.

You can refer to this snippet for the same: https://github.com/juspay/hyperswitch-card-vault/blob/86cf0ffe2eb538676ba2940dd77200d4f31bc935/.github/workflows/CI.yml#L15-L31